aait/backend/group-2B/Application.Tests/Features/Authentication/Handlers/Commands/RegisterUserCommandHandlerTest.cs

```diff
@@ -117,13 +121,124 @@ public async Task Invalid_DuplicateEmailRegisterNewUsew()
             CancellationToken.None
         );
 
+        // Change the username (username and email need to be unique)
+        newUser.Username = "new_username";
+
+        var result = await _handler.Handle(
+            new RegisterUserCommand { RegisterUserDto = newUser },
+            CancellationToken.None
+        );
+
+        result.ShouldBeOfType<CommonResponse<LoggedInUserDto>>();
+        result.IsSuccess.ShouldBeFalse();
+        result.Error.ShouldBe(new[] { "Email Exists." });
+        result.Value.ShouldBeNull();
+    }
+
+    [Fact]
+    public async Task Invalid_DuplicateUsernameRegisterNewUsew()
+    {
+        var newUser = new RegisterUserDto
+        {
+            FirstName = "User4",
+            LastName = "User4",
+            Bio = "user bio",
+            Email = "User4@gmail.com",
+            Username = "User4",
+            Password = "User4password",
+            Phone = "1234567890",
+        };
+
+        await _handler.Handle(
+            new RegisterUserCommand { RegisterUserDto = newUser },
+            CancellationToken.None
+        );
+
+        newUser.Email = "newemail@email.com";
+
```

Comment:
You don't need to call the handler twice . You can use the user in the user mockreposiotry